src/fidesops/task/filter_results.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from typing import List, Dict, Any, Union
+
+from fidesops.graph.config import FieldPath
+
+logger = logging.getLogger(__name__)
+
+
+def select_and_save_field(saved: Any, row: Any, target_path: FieldPath) -> Dict:
+    """Extract the data located along the given `target_path` from the row and add to the "saved" dictionary.
+
+    Entire rows are returned from your collections; this function will incrementally just pull the PII from the rows,
+    by retrieving data along target_paths to relevant data categories.
+
+    To use, you might start with a "saved" that is an empty dict, and loop through a list of FieldPaths you want,
+    continuing to pass in the ever-growing new "saved" collection that was returned from the previous iteration.
+
+    :param saved: Call with an empty dict to start, it will recursively update as data along the target_path is added to it.
+    :param row: Call with retrieved row to start, it will recursively be called with a variety of object types until we
+    reach the most deeply nested value.
+    :param target_path: FieldPath to the data we want to retrieve
+
+    :return: modified saved dictionary with given field path if found
+    """
+
+    def _defaultdict_or_array(resource: Any) -> Any:
+        """Helper for building new nested resource - can return an empty dict, empty array or resource itself"""
+        return type(resource)() if isinstance(resource, (list, dict)) else resource
+
+    if isinstance(row, list):
+        for i, elem in enumerate(row):
+            try:
+                saved[i] = select_and_save_field(saved[i], elem, target_path)
+            except IndexError:
+                saved.append(
+                    select_and_save_field(
+                        _defaultdict_or_array(elem), elem, target_path
+                    )
+                )
+
+    elif isinstance(row, dict):
+        for key in row:
+            if key == target_path.levels[0]:
+                if key not in saved:
+                    saved[key] = _defaultdict_or_array(row[key])
+                saved[key] = select_and_save_field(
+                    saved[key], row[key], FieldPath(*target_path.levels[1:])
+                )
+    return saved
+
+
+RecursiveRow = Union[Dict[Any, Any], List[Any]]
+
+
+def remove_empty_objects(row: RecursiveRow) -> RecursiveRow:
```

Comment:
Why aren't we removing empty arrays too?

Reply:
An earlier commit removed the arrays, and I reverted it because my primary reason for removing empty dicts is because I created them. 

There are cases when I can preemptively add dicts to the results in `select_and_save_field`, expecting data to be retrieved in that row.  However, sometimes a row may not have data on a field path, so I created an empty dict for no reason.  I am trying to delete dicts that I created.   I don't think I am creating empty arrays here, so I didn't want to delete an empty array that was in the user data.  Although I can remove it too, it's probably not adding anything, this would be mostly restoring an old commit.

Reply:
OK removing empty arrays now, this also has a cascading effect, removing an empty dict or empty array can cause higher levels in the path to be removed as well (this was already happening).